### ISSUE TITLE
barcodes path unavailable in conda install

### ISSUE THREAD

**Issue by cachris1** [Lexogen-Tools/idemuxcpp#1]
After installing with conda and running, I got this error: 
`Error: the path /usr/share/idemuxcpp/barcodes/i1 is not available!`
It seems with conda the barcodes are by default located at: 

`~/miniconda3/envs/quantseq/share/idemuxcpp/`

conda info:
```
     active environment : quantseq
    active env location : /home/cachris/miniconda3/envs/quantseq
            shell level : 2
       user config file : /home/cachris/.condarc
 populated config files :
          conda version : 23.3.1
    conda-build version : not installed
         python version : 3.10.11.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.35=0
                          __linux=5.15.0=0
                          __unix=0=0
       base environment : /home/cachris/miniconda3  (writable)
      conda av data dir : /home/cachris/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/cachris/miniconda3/pkgs
                          /home/cachris/.conda/pkgs
       envs directories : /home/cachris/miniconda3/envs
                          /home/cachris/.conda/envs
               platform : linux-64
             user-agent : conda/23.3.1 requests/2.28.1 CPython/3.10.11 Linux/5.15.0-71-generic ubuntu/22.04.1 glibc/2.35
                UID:GID : 1045:1079
             netrc file : None
           offline mode : False

```
conda list:

```

# packages in environment at /home/cachris/miniconda3/envs/quantseq:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
bamtools                  2.5.1               hd03093a_10    bioconda
bcrypt                    3.2.0            py36h7b6447c_0
blas                      1.0                         mkl
boost-cpp                 1.78.0               h6582d0a_3    conda-forge
brotlipy                  0.7.0           py36h27cfd23_1003
bzip2                     1.0.8                h7f98852_4    conda-forge
ca-certificates           2023.7.22            hbcca054_0    conda-forge
certifi                   2021.5.30        py36h06a4308_0
cffi                      1.14.6           py36h400218f_0
charset-normalizer        2.0.4              pyhd3eb1b0_0
cryptography              35.0.0           py36hd23ed53_0
dataclasses               0.8                pyh4f3eec9_6
dbus                      1.13.18              hb2f20db_0
docutils                  0.17.1           py36h06a4308_1
dropbox                   5.2.1                    py36_0    bioconda
expat                     2.4.9                h6a678d5_0
fastqc                    0.11.8                        0    bioconda
filechunkio               1.6                      py36_0    bioconda
freetype                  2.12.1               h4a9f257_0
ftputil                   3.2                      py36_0    bioconda
gdbm                      1.18                 hd4cb3f1_4
glib                      2.69.1               h4ff587b_1
icu                       72.1                 hcb278e6_0    conda-forge
idemux                    0.1.6              pyhdfd78af_0    bioconda
idemuxcpp                 0.1.9                h2d38e66_2    bioconda
idna                      3.3                pyhd3eb1b0_0
importlib_resources       5.2.0              pyhd3eb1b0_1
intel-openmp              2022.1.0          h9e868ea_3769
ld_impl_linux-64          2.38                 h1181459_1
libffi                    3.3                  he6710b0_2
libgcc-ng                 13.1.0               he5830b7_0    conda-forge
libgomp                   13.1.0               he5830b7_0    conda-forge
libpng                    1.6.39               h5eee18b_0
libsodium                 1.0.18               h7b6447c_0
libstdcxx-ng              13.1.0               hfd8a6a1_0    conda-forge
libxcb                    1.15                 h7f8727e_0
libzlib                   1.2.13               hd590300_5    conda-forge
mkl                       2020.2                      256
mkl-service               2.3.0            py36he8ac12f_0
mkl_fft                   1.3.0            py36h54f3939_0
mkl_random                1.1.1            py36h0573a6f_0
ncurses                   6.4                  h6a678d5_0
numpy                     1.19.2           py36h54aff64_0
numpy-base                1.19.2           py36hfa32c7d_0
openjdk                   11.0.13              h87a67e3_0
openssl                   1.1.1v               h7f8727e_0
pandas                    1.1.5            py36ha9443f7_0
paramiko                  2.8.1              pyhd3eb1b0_0
pcre                      8.45                 h295c915_0
perl                      5.34.0               h5eee18b_2
pip                       21.2.2           py36h06a4308_0
psutil                    5.8.0            py36h27cfd23_1
pycparser                 2.21               pyhd3eb1b0_0
pynacl                    1.4.0            py36h7b6447c_1
pyopenssl                 22.0.0             pyhd3eb1b0_0
pysftp                    0.2.9                    py36_0    bioconda
pysocks                   1.7.1            py36h06a4308_0
python                    3.6.13               h12debd9_1
python-dateutil           2.8.2              pyhd3eb1b0_0
pytz                      2021.3             pyhd3eb1b0_0
pyyaml                    5.4.1            py36h27cfd23_1
readline                  8.2                  h5eee18b_0
requests                  2.27.1             pyhd3eb1b0_0
setuptools                58.0.4           py36h06a4308_0
six                       1.16.0             pyhd3eb1b0_1
snakemake                 3.13.3                   py36_0    bioconda
sqlite                    3.41.2               h5eee18b_0
tk                        8.6.12               h1ccaba5_0
tqdm                      4.7.2                    py36_0    bioconda
urllib3                   1.26.8             pyhd3eb1b0_0
wheel                     0.37.1             pyhd3eb1b0_0
wrapt                     1.12.1           py36h7b6447c_1
xz                        5.2.6                h166bdaf_0    conda-forge
yaml                      0.2.5                h7b6447c_0
zipp                      3.6.0              pyhd3eb1b0_0
zlib                      1.2.13               hd590300_5    conda-forge
zstd                      1.5.2                hfc55251_7    conda-forge

```

**Comment by entzian** [comment]
Dear @cachris1,
thanks for opening this issue! We will improve loading the barcode maps in the next version.
In the current version, you can use the command line parameter `-m, --correction-map-prefix=STRING` or in your case  `-m ~/miniconda3/envs/quantseq/share/idemuxcpp/barcodes` . In general the barcodes main directory can be found with `find / -wholename "*idemuxcpp/barcodes" -type d`.

**Comment by cachris1** [comment]
Thanks for the reply, I will try that out.

**Comment by freekvh** [comment]
I know this issue is closed but I am on version 0.3.0, which came out after this issue was filed, and I am still getting this error:

```bash
# idemuxCPP --r1 /data/NN1_CD14_S63_L001_R1_001.fastq.gz --r2 /data/NN1_CD14_S63_L001_R2_001.fastq.gz --sample-sheet /experiment/sample_sheet.csv --out /experiment/demultiplexed
idemuxCPP
Trying to find the appropriate barcode set for i7...
No barcodes have been specified for i7.
Trying to find the appropriate barcode set for i5...
No barcodes have been specified for i5.
Trying to find the appropriate barcode set for i1...
terminate called after throwing an instance of 'std::runtime_error'
  what():  Error: the path /usr/share/idemuxcpp/barcodes is not available!

Aborted (core dumped)
```

Indeed the `-m` parameter solves this, in my container environment I found the barcodes under `/opt/Miniforge3/pkgs/idemuxcpp-0.3.0-h077b44d_2/share/idemuxcpp/barcodes`

Best regards.
